test/e2e/e2e_helpers_test.go

```diff
@@ -421,10 +422,91 @@ func sortNamespacedNames(nns []types.NamespacedName) {
 	})
 }
 
+// Note: This method is currently not being used in the E2E test.
+/*
 func testDataPath(description ginkgo.GinkgoTestDescription) string {
 	path := filepath.Join(description.ComponentTexts...)
 	return strings.ReplaceAll(path, " ", "_")
 }
+*/
+
+// Get unstructured object with Kuberentes dynamic clients.
+func GetUnstructuredObject(ctx context.Context, d dynamic.Interface, gvr schema.GroupVersionResource,
+	resource types.NamespacedName) (*unstructured.Unstructured, error) {
+	var (
+		unstructuredObject *unstructured.Unstructured
+		err                error
+	)
+
+	unstructuredObject, err = d.Resource(gvr).Namespace(resource.Namespace).Get(ctx, resource.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return unstructuredObject, nil
+}
+
+// Get Deployment object with Kubernetes typed clients.
+func GetDeployment(ctx context.Context, c client.Client, resource types.NamespacedName) (*appsv1.Deployment, error) {
+	dep := &appsv1.Deployment{}
+
+	err := c.Get(ctx, resource, dep)
+	if err != nil {
+		return dep, err
+	}
+
+	return dep, nil
+}
+
+// Get Service object with Kubernetes typed clients.
+func GetService(ctx context.Context, c client.Client, resource types.NamespacedName) (*corev1.Service, error) {
+	svc := &corev1.Service{}
+
+	err := c.Get(ctx, resource, svc)
+	if err != nil {
+		return svc, err
+	}
+
+	return svc, nil
+}
+
+// Get a container list of given Deployment object.
+func GetContainersFromDeployment(dep *appsv1.Deployment) []corev1.Container {
+	return dep.Spec.Template.Spec.Containers
+}
+
+// Validate if the container exists in given container list.
+func ContainerExists(containerList []corev1.Container, containerName string) error {
```

Comment:
Please unexport the functions where possible:
```suggestion
func containerExists(containerList []corev1.Container, containerName string) error {
```

Reply:
Thanks for the comment. Addressed it by make those functions un-exported.